### PR TITLE
MOBILE-391: Document + Fix JS takeOff

### PR DESCRIPTION
**Pull request by rlepinski**
### What do these changes do?
Updates the readme with JS takeOff and configure steps. Also fixed a takeOff error log due to calling takeOff on a bg thread on Android.

### Why are these changes necessary?
People want to control everything through JS instead of xml

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,88 +29,112 @@ Please visit http://support.urbanairship.com/ for any issues integrating or usin
 
         cordova plugin add urbanairship-cordova
 
+2. *(Android Only)* Add a reference to your google-servies.json file (parallel to config.xml):
```
Comment (BrianBatchelder) [inline README.md]: Does `parallel` mean `in the same directory as`? If so, maybe that would be clearer?

File: README.md [hunk 2]
```diff
@@ -29,88 +29,112 @@ Please visit http://support.urbanairship.com/ for any issues integrating or usin
...
+                    "developmentTeam": "XXXXXXXXXX"
+                },
+                "release": {
+                    "developmentTeam": "XXXXXXXXXX"
+                }
+            }
+        }
+    Your iOS builds will need to reference the build.json using Cordova's "--buildConfig" flag.
+
+4. Initailize Urban Airship
```
Comment (BrianBatchelder) [inline README.md]: _Initialize_

File: README.md [hunk 3]
```diff
@@ -29,88 +29,112 @@ Please visit http://support.urbanairship.com/ for any issues integrating or usin
...
+          accentColor: "#FF0000"
+        })
+
+        // Configure iOS
+        UAirship.setPresentationOptions(
+          UAirship.presentationOptions.sound | UAirship.presentationOptions.alert
+        )
+
+
+    Alternatively you can configure Urban Airship through config.xml and it will takeOff automatically.
```
Comment (BrianBatchelder) [inline README.md]: Is JS now our recommended takeoff mechanism? If not, this line should probably be before the JS lines.

File: README.md [hunk 4]
```diff
@@ -29,88 +29,112 @@ Please visit http://support.urbanairship.com/ for any issues integrating or usin
...
-
         <!-- Urban Airship development log level defaults to debug -->
         <preference name="com.urbanairship.development_log_level" value="none | error | warn | info | debug | verbose" />
 
         <!-- Urban Airship production log level defaults to error -->
         <preference name="com.urbanairship.production_log_level" value="none | error | warn | info | debug | verbose" />
 
-        <!-- Override the Android notification icon -->
-        <preference name="com.urbanairship.notification_icon" value="ic_notification" />
+        <!-- Enables/disables auto launching the message center when the corresponding push is opened. -->
```
Comment (BrianBatchelder) [inline README.md]: Should you indicate what happens if the value is not set, as you did for the `in_production` flag?

File: README.md [hunk 5]
```diff
@@ -29,88 +29,112 @@ Please visit http://support.urbanairship.com/ for any issues integrating or usin
...
-                "debug": {
-                    "developmentTeam": "XXXXXXXXXX"
-                },
-                "release": {
-                    "developmentTeam": "XXXXXXXXXX"
-                }
-            }
-        }
-    Your iOS builds will need to reference the build.json using Cordova's "--buildConfig" flag.
+    `UrbanAirship.takeOff` can be called multiple times but any changes the app credentials will not apply until the next app start.
```
Comment (BrianBatchelder) [inline README.md]: any changes _to_ the app credentials

File: README.md [hunk 6]
```diff
@@ -29,88 +29,112 @@ Please visit http://support.urbanairship.com/ for any issues integrating or usin
...
-4. Enable user notifications:
+5. Enable user notifications:
 
-        // Enable user notifications (will prompt the user to accept push notifications)
+        // Enable user notifications (will prompt the user to accept push notifications on iOS)
         UAirship.setUserNotificationsEnabled(true, function (enabled) {
             console.log("User notifications are enabled! Fire away!")
         })
 
+6. Listen for events:
```
Comment (BrianBatchelder) [inline README.md]: Maybe indicate if this step is optional or required?

File: README.md (hunk 4 above)
Comment (rlepinski) [reply README.md]: Going to wait on describing this further. We need to add a feature from react-native before this is really that useful. Right now cordova users need to parse the push message for the message ID. Its going to be differrent per platform. On React we send a special even when its disabled - https://docs.urbanairship.com/reference/libraries/react-native/latest/UrbanAirship.html#event:showInbox

File: README.md [hunk 7]
```diff
@@ -29,88 +29,111 @@ Please visit http://support.urbanairship.com/ for any issues integrating or usin
 
         cordova plugin add urbanairship-cordova
 
+2. *(Android Only)* Add a reference to your google-servies.json file in the app's `config.xml`:
```
Comment (BrianBatchelder) [inline README.md]: _google-services.json_

File: README.md (hunk 3 above)
Comment (BrianBatchelder) [reply README.md]: OK

File: README.md (hunk 4 above)
Comment (BrianBatchelder) [reply README.md]: OK

**Comment by crow** [comment]
🍔 sure

**Pull request merged**